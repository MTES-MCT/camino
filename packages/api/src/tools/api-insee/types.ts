export interface IApiSirenUniteLegalePeriode
  extends IApiSirenUnionUniteLegalePeriodeEtablissmentUnite {
  dateDebut: Date
  dateFin: Date
}

export interface IApiSirenUnionUniteLegalePeriodeEtablissmentUnite {
  nicSiegeUniteLegale: string | null
  denominationUniteLegale: string | null
  denominationUsuelle1UniteLegale: string | null
  nomUniteLegale: string | null
}

export interface IApiSirenUnionUniteLegaleEtablissmentUnite {
  prenomUsuelUniteLegale: string
  sexeUniteLegale: 'F' | 'M' | null
  sigleUniteLegale: string | null
}

export interface IApiSirenEtablissementUnite
  extends IApiSirenUnionUniteLegalePeriodeEtablissmentUnite,
    IApiSirenUnionUniteLegaleEtablissmentUnite {
  categorieEntreprise: string
  categorieJuridiqueUniteLegale: string
  dateCreationUniteLegale: Date | null
}

export interface IApiSirenEtablissementAdresse {
  codeCedexEtablissement: string | null
  codePaysEtrangerEtablissement: string | null
  numeroVoieEtablissement: string | null
  indiceRepetitionEtablissement: string | null
  typeVoieEtablissement: string | null
  libelleVoieEtablissement: string | null
  codePostalEtablissement: string | null
  libelleCommuneEtablissement: string
  libelleCommuneEtrangerEtablissement: string
}

export interface IApiSirenEtablissement {
  adresseEtablissement: IApiSirenEtablissementAdresse
  siren: string
  uniteLegale: IApiSirenEtablissementUnite
}

export interface IApiSirenUniteLegale
  extends IApiSirenUnionUniteLegaleEtablissmentUnite {
  siren: string
  periodesUniteLegale: IApiSirenUniteLegalePeriode[]
}

export interface IApiSirenQuery {
  fault?: {
    code: number
    description: string
    message: string
  }
  error?: {
    // eslint-disable-next-line camelcase
    error_description: string
  }
  header: {
    debut: number
    nombre: number
    message: 'OK' | string
    statut: number
    total: number
  }
}

export interface IApiSirenQueryTypes extends IApiSirenQuery {
  etablissements?: IApiSirenEtablissement[]
  unitesLegales?: IApiSirenUniteLegale[]
}

export interface IApiSirenQueryToken extends IApiSirenQuery {
  result?: {
    // eslint-disable-next-line camelcase
    access_token: string
  }
}
