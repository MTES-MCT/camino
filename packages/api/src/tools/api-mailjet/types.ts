export enum EmailTemplateId {
  DEMARCHE_CONFIRMATION_DEPOT = 3413770,
  ACTIVITES_NOUVELLES = 3794302,
  ACTIVITES_RELANCE = 3785214,
}

export const EmailAdministration = {
  ONF: 'pole.minier@onf.fr',
  PTMG: 'ptmg@ctguyane.fr',
  DGTM: 'mc.remd.deal-guyane@developpement-durable.gouv.fr',
} as const
