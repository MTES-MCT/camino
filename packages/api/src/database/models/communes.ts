import { Model } from 'objection'

import { ICommune } from '../../types'

interface Communes extends ICommune {}

class Communes extends Model {
  public static tableName = 'communes'

  public static jsonSchema = {
    type: 'object',
    required: ['id', 'nom'],

    properties: {
      id: { type: 'string', maxLength: 8 },
      nom: { type: 'string' },
      departementId: { type: 'string' }
    }
  }
}

export default Communes
