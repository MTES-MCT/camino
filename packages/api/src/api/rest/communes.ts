import { Commune, communeIdValidator } from 'camino-common/src/static/communes.js'
import { Pool } from 'pg'
import { HTTP_STATUS } from 'camino-common/src/http.js'
import { CaminoRequest, CustomResponse } from './express-type.js'
import { getCommunes as getCommunesQuery } from '../../database/queries/communes.queries.js'
import { z } from 'zod'

export const getCommunes = (pool: Pool) => async (req: CaminoRequest, res: CustomResponse<Commune[]>) => {
  try {
    const communeIds = z.array(communeIdValidator).nonempty().safeParse(req.query.ids)
    if (communeIds.success) {
      const communes = await getCommunesQuery(pool, { ids: communeIds.data })
      res.json(communes)
    } else {
      res.sendStatus(HTTP_STATUS.HTTP_STATUS_BAD_REQUEST)
    }
  } catch (e) {
    console.error(e)

    res.sendStatus(HTTP_STATUS.HTTP_STATUS_INTERNAL_SERVER_ERROR)
  }
}
