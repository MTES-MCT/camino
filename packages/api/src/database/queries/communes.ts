import { dbQueryAndValidate } from '../../pg-database.js'
import { CommuneId, communeValidator } from 'camino-common/src/static/communes.js'
import { Pool } from 'pg'
import { getCommunes } from './communes.queries.js'
import { isNonEmptyArray } from 'camino-common/src/typescript-tools.js'

export const getCommunesIndex = async (pool: Pool, ids: CommuneId[]): Promise<Record<CommuneId, string>> => {
  const communes = isNonEmptyArray(ids) ? await dbQueryAndValidate(getCommunes, { ids }, pool, communeValidator) : []

  return communes.reduce<Record<CommuneId, string>>((acc, commune) => {
    acc[commune.id] = commune.nom

    return acc
  }, {})
}
