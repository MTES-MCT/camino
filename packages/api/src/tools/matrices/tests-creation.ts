import '../../init'
import { writeFileSync } from 'fs'

import { anneePrecedente, caminoAnneeValidator, toCaminoAnnee } from 'camino-common/src/date'
import { titresGet } from '../../database/queries/titres'
import { titresActivitesGet } from '../../database/queries/titres-activites'
import { userSuper } from '../../database/user-super'
import { REGION_IDS } from 'camino-common/src/static/region'
import { getEntreprises, getEntreprisesValidor } from '../../api/rest/entreprises.queries'
import { config } from '../../config'
import pg from 'pg'
import { z } from 'zod'
import { TitreId, titreIdValidator } from 'camino-common/src/validators/titres'
import { EntrepriseId, entrepriseIdValidator } from 'camino-common/src/entreprise'
import { substanceLegaleIdValidator } from 'camino-common/src/static/substancesLegales'
import { CommuneId, communeIdValidator } from 'camino-common/src/static/communes'
import { idGenerate, newTitreId } from '../../database/models/_format/id-create'
import { RawLineMatrice, getRawLines } from '../../business/matrices'
// Le pool ne doit Ãªtre qu'aux entrypoints : le daily, le monthly, et l'application.
const pool = new pg.Pool({
  host: config().PGHOST,
  user: config().PGUSER,
  password: config().PGPASSWORD,
  database: config().PGDATABASE,
})

const titreIdsIndex: Record<TitreId, TitreId> = {}

const titreIdTransformer = titreIdValidator.transform((value: TitreId): TitreId => {
  if (titreIdsIndex[value] === undefined) {
    const newId = newTitreId()
    titreIdsIndex[value] = newId
  }

  return titreIdsIndex[value]
})

const entrepriseIdsIndex: Record<EntrepriseId, EntrepriseId> = {}
const communeIdsIndex: Record<CommuneId, CommuneId> = {}

const entrepriseIdTransformer = entrepriseIdValidator.transform((value: EntrepriseId): EntrepriseId => {
  if (entrepriseIdsIndex[value] === undefined) {
    const newId = idGenerate(8) as EntrepriseId
    entrepriseIdsIndex[value] = newId
  }

  return entrepriseIdsIndex[value]
})
const contenuStripedValidator = z.object({
  substancesFiscales: z.any().optional(),
  renseignements: z
    .object({
      environnement: z.any(),
    })
    .optional(),
})
const entryValidator = z.object({
  activitesAnnuelles: z.array(
    z.object({
      titreId: titreIdTransformer,
      contenu: contenuStripedValidator,
    })
  ),
  activitesTrimestrielles: z.array(
    z.object({
      titreId: titreIdTransformer,
      contenu: contenuStripedValidator,
    })
  ),
  titres: z.array(
    z.object({
      titulaireIds: z.array(entrepriseIdTransformer),
      amodiataireIds: z.array(entrepriseIdTransformer),
      substances: z.array(substanceLegaleIdValidator),
      communes: z.array(
        z.object({
          id: communeIdValidator.transform(id => {
            if (communeIdsIndex[id] === undefined) {
              const newId = `973${idGenerate(2)}` as CommuneId
              communeIdsIndex[id] = newId
            }

            return communeIdsIndex[id]
          }),
          surface: z.number(),
        })
      ),
      id: titreIdTransformer,
    })
  ),
  annee: caminoAnneeValidator,
  entreprises: z.array(getEntreprisesValidor),
})

export type BodyMatrice = {
  entries: z.infer<typeof entryValidator>
  expected: RawLineMatrice[]
}
const writeMatricesForTest = async () => {
  const user = userSuper
  const testBody: BodyMatrice[] = []
  const entreprises = await getEntreprises(pool)

  const annees = [toCaminoAnnee(2023), toCaminoAnnee(2022)] as const
  for (const annee of annees) {
    const anneeMoins1 = anneePrecedente(annee)

    const titres = await titresGet(
      { regions: [REGION_IDS.Guyane] },
      {
        fields: {
          titulairesEtape: { id: {} },
          amodiatairesEtape: { id: {} },
          substancesEtape: { id: {} },
          pointsEtape: { id: {} },
        },
      },
      user
    )

    const activites = await titresActivitesGet(
      {
        typesIds: ['grx', 'gra', 'wrp'],
        statutsIds: ['dep'],
        annees: [anneeMoins1],
        titresIds: titres.map(({ id }) => id),
      },
      { fields: { id: {} } },
      user
    )
    const activitesTrimestrielles = await titresActivitesGet(
      {
        typesIds: ['grp'],
        statutsIds: ['dep'],
        annees: [anneeMoins1],
        titresIds: titres.map(({ id }) => id),
      },
      { fields: { id: {} } },
      user
    )
    const stripedData = entryValidator.parse({
      activitesAnnuelles: activites,
      activitesTrimestrielles,
      titres,
      annee,
      entreprises,
    })

    const result = getRawLines(
      stripedData.activitesAnnuelles,
      stripedData.activitesTrimestrielles,
      stripedData.titres,
      annee,
      Object.values(communeIdsIndex).map(communeId => ({ id: communeId, nom: communeId })),
      stripedData.entreprises
    )
    testBody.push({
      entries: stripedData,
      expected: result,
    })
  }
  writeFileSync(`src/business/matrices.cas.json`, JSON.stringify(testBody))
}

writeMatricesForTest()
  .then(() => {
    process.exit()
  })
  .catch(e => {
    console.error(e)
    process.exit(1)
  })
