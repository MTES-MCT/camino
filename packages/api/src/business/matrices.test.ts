import { buildMatrices } from './matrices.js'
import { ITitre } from '../types.js'
import { newEntrepriseId } from 'camino-common/src/entreprise.js'
import { describe, expect, test } from 'vitest'
import { checkCodePostal } from 'camino-common/src/static/departement.js'
import { toCommuneId } from 'camino-common/src/static/communes.js'
describe('matrices', () => {
  test('buildMatrices', () => {
    const openFiscaResponse = {
      articles: {
        'titre1-auru-97310': {
          surface_communale: { '2020': 19805494 },
          surface_communale_proportionnee: { '2020': 1 },
          quantite_aurifere_kg: { '2020': 12.243 },
          redevance_communale_des_mines_aurifere_kg: { '2021': 284.71 },
          redevance_departementale_des_mines_aurifere_kg: { '2021': 56.27 },
          taxe_guyane_brute: { '2021': 852.29 },
          taxe_guyane_deduction: { '2021': 200 },
          taxe_guyane: { '2021': 652.29 },
        },
        'titre2-auru-97358': {
          surface_communale: { '2020': 323907 },
          surface_communale_proportionnee: { '2020': 0.3191007 },
          quantite_aurifere_kg: { '2020': 3.1591 },
          redevance_communale_des_mines_aurifere_kg: { '2021': 167.64 },
          redevance_departementale_des_mines_aurifere_kg: { '2021': 33.47 },
          taxe_guyane_brute: { '2021': 502.08 },
          taxe_guyane_deduction: { '2021': 0 },
          taxe_guyane: { '2021': 502.08 },
        },
        'titre2-auru-97312': {
          surface_communale: { '2020': 691155 },
          surface_communale_proportionnee: { '2020': 0.6808993 },
          quantite_aurifere_kg: { '2020': 3.1591 },
          redevance_communale_des_mines_aurifere_kg: { '2021': 357.72 },
          redevance_departementale_des_mines_aurifere_kg: { '2021': 71.41 },
          taxe_guyane_brute: { '2021': 1071.34 },
          taxe_guyane_deduction: { '2021': 0 },
          taxe_guyane: { '2021': 1071.34 },
        },
        'titre3-auru-97311': {
          surface_communale: { '2020': 1005053 },
          surface_communale_proportionnee: { '2020': 1 },
          quantite_aurifere_kg: { '2020': 6.118269999999995 },
          redevance_communale_des_mines_aurifere_kg: { '2021': 933.47 },
          redevance_departementale_des_mines_aurifere_kg: { '2021': 186.13 },
          taxe_guyane_brute: { '2021': 17950.27 },
          taxe_guyane_deduction: { '2021': 5000 },
          taxe_guyane: { '2021': 12950.27 },
        },
      },
      titres: {
        titre1: {
          commune_principale_exploitation: { '2020': '97310' },
          surface_totale: { '2020': 19805494 },
          operateur: { '2020': 'titulaire1' },
          investissement: { '2020': '131535' },
          categorie: { '2020': 'pme' },
          articles: ['titre1-auru-97310'],
        },
        titre2: {
          commune_principale_exploitation: { '2020': '97312' },
          surface_totale: { '2020': 1015062 },
          operateur: { '2020': 'titulaire2' },
          investissement: { '2020': '0' },
          categorie: { '2020': 'pme' },
          articles: ['titre2-auru-97358', 'titre2-auru-97312'],
        },
        titre3: {
          commune_principale_exploitation: { '2020': '97311' },
          surface_totale: { '2020': 1005053 },
          operateur: { '2020': 'titulaire3' },
          investissement: { '2020': '28000' },
          categorie: { '2020': 'pme' },
          articles: ['titre3-auru-97311'],
        },
      },
      communes: {
        '97310': { articles: ['titre1-auru-97310'] },
        '97311': { articles: ['titre3-auru-97311'] },
        '97312': { articles: ['titre2-auru-97312'] },
        '97358': { articles: ['titre2-auru-97358'] },
      },
    }
    const titres: Pick<ITitre, 'id' | 'slug' | 'titulaires' | 'communes'>[] = [
      {
        id: 'titre1',
        titulaires: [
          {
            id: newEntrepriseId(''),
            nom: 'titulaire1',
            adresse: 'ladresse1',
            legalSiren: 'legalSiren1',
          },
        ],
        slug: 'slug-titre-1',
        communes: [
          {
            id: toCommuneId('97310'),
            nom: 'Roura',
          },
        ],
      },
      {
        id: 'titre2',
        titulaires: [
          {
            id: newEntrepriseId(''),
            nom: 'titulaire2',
            adresse: 'ladresse2',
            legalSiren: 'legalSiren2',
          },
        ],
        slug: 'slug-titre-2',
        communes: [
          {
            id: toCommuneId('97358'),
            nom: 'Saint-Ã‰lie',
          },
          {
            id: toCommuneId('97312'),
            nom: 'Sinnamary',
          },
        ],
      },
      {
        id: 'titre3',
        titulaires: [
          {
            id: newEntrepriseId(''),
            nom: 'titulaire3',
            adresse: 'ladresse3',
            codePostal: checkCodePostal('97311'),
            commune: 'Saint-Laurent-du-Maroni',
            legalSiren: 'legalSiren3',
          },
        ],
        slug: 'slug-titre-3',
        communes: [
          {
            id: toCommuneId('97311'),
            nom: 'Saint-Laurent-du-Maroni',
          },
        ],
      },
    ]

    expect(
      buildMatrices(openFiscaResponse, titres, 2021, {
        substances: {
          aloh: { tarifCommunal: 636.4, tarifDepartemental: 127.1 },
          anti: { tarifCommunal: 13.1, tarifDepartemental: 3 },
          arge: { tarifCommunal: 270.2, tarifDepartemental: 53.8 },
          arse: { tarifCommunal: 731.7, tarifDepartemental: 148.7 },
          auru: { tarifCommunal: 166.3, tarifDepartemental: 33.2 },
          bism: { tarifCommunal: 64.1, tarifDepartemental: 13 },
          cfxa: { tarifCommunal: 254.1, tarifDepartemental: 123.4 },
          cfxb: { tarifCommunal: 972.9, tarifDepartemental: 192.6 },
          cfxc: { tarifCommunal: 236.3, tarifDepartemental: 52.4 },
          coox: { tarifCommunal: 356.2, tarifDepartemental: 72.8 },
          cuiv: { tarifCommunal: 21.3, tarifDepartemental: 4.3 },
          etai: { tarifCommunal: 132.5, tarifDepartemental: 26.2 },
          fera: { tarifCommunal: 545.3, tarifDepartemental: 111.4 },
          ferb: { tarifCommunal: 375.7, tarifDepartemental: 78.1 },
          fluo: { tarifCommunal: 827.3, tarifDepartemental: 168.1 },
          hyda: { tarifCommunal: 1635.1, tarifDepartemental: 325.4 },
          hydb: { tarifCommunal: 54.5, tarifDepartemental: 11.3 },
          hydc: { tarifCommunal: 1221.2, tarifDepartemental: 1569.1 },
          hydd: { tarifCommunal: 9.3, tarifDepartemental: 7.3 },
          hyde: { tarifCommunal: 8.6, tarifDepartemental: 6.5 },
          hydf: { tarifCommunal: 351.4, tarifDepartemental: 513 },
          kclx: { tarifCommunal: 286.6, tarifDepartemental: 57.1 },
          lith: { tarifCommunal: 54.5, tarifDepartemental: 11.2 },
          mang: { tarifCommunal: 406.3, tarifDepartemental: 82.2 },
          moly: { tarifCommunal: 270.2, tarifDepartemental: 54.5 },
          naca: { tarifCommunal: 786.7, tarifDepartemental: 159.9 },
          nacb: { tarifCommunal: 478.9, tarifDepartemental: 94.4 },
          nacc: { tarifCommunal: 159.9, tarifDepartemental: 31 },
          plom: { tarifCommunal: 686.4, tarifDepartemental: 132.5 },
          souf: { tarifCommunal: 3.1, tarifDepartemental: 1.7 },
          uran: { tarifCommunal: 323.7, tarifDepartemental: 64.3 },
          wolf: { tarifCommunal: 148.7, tarifDepartemental: 29.1 },
          zinc: { tarifCommunal: 545.3, tarifDepartemental: 111.4 },
        },
        tarifTaxeMinierePME: 498.06,
        tarifTaxeMiniereAutre: 123.97,
      })
    ).toMatchSnapshot()
  })
})
