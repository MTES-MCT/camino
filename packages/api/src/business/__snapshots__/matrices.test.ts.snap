// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matrices buildMatrices 1`] = `
Object {
  "matrice1121": Array [
    Object {
      "Base des redevances | Nature": "Kilogramme d'or contenu",
      "Base des redevances | Quantités": "12.243",
      "Commune du lieu principal d'exploitation": "Roura",
      "Désignation et adresse des concessionnaires, titulaires de permis d’exploitation ou exploitants": "titulaire1 - ladresse1 - legalSiren1 SIREN",
      "Frais de gestion de la fiscalité directe locale": 79.4616,
      "Nature des substances extraites": "Minerais aurifères",
      "Numéro d'ordre de la matrice": 1,
      "Numéro de l'article du rôle": "slug-titre-1",
      "Redevance communale | Montant net redevance des mines": 284.71,
      "Redevance communale | Tarifs": 166.3,
      "Redevance départementale | Montant net": 56.27,
      "Redevance départementale | Tarifs": 33.2,
      "Service de la Direction générale des Finances publiques en charge du recouvrement": "Direction régionale des finances publiques (DRFIP) - Guyane",
      "Taxe minière sur l'or de Guyane | Montant des investissements déduits": 200,
      "Taxe minière sur l'or de Guyane | Montant net de taxe minière sur l'or de Guyane": 652.29,
      "Taxe minière sur l'or de Guyane | Tarifs par kg extrait pour les PME": 498.06,
      "Taxe minière sur l'or de Guyane | Tarifs par kg extrait pour les autres entreprises": 0,
      "Total redevance des mines": 340.97999999999996,
    },
    Object {
      "Base des redevances | Nature": "Kilogramme d'or contenu",
      "Base des redevances | Quantités": "1.008",
      "Commune du lieu principal d'exploitation": "Sinnamary",
      "Désignation et adresse des concessionnaires, titulaires de permis d’exploitation ou exploitants": "titulaire2 - ladresse2 - legalSiren2 SIREN",
      "Frais de gestion de la fiscalité directe locale": 56.255199999999995,
      "Nature des substances extraites": "Minerais aurifères",
      "Numéro d'ordre de la matrice": 2,
      "Numéro de l'article du rôle": "slug-titre-2",
      "Redevance communale | Montant net redevance des mines": 167.64,
      "Redevance communale | Tarifs": 166.3,
      "Redevance départementale | Montant net": 33.47,
      "Redevance départementale | Tarifs": 33.2,
      "Service de la Direction générale des Finances publiques en charge du recouvrement": "Direction régionale des finances publiques (DRFIP) - Guyane",
      "Taxe minière sur l'or de Guyane | Montant des investissements déduits": 0,
      "Taxe minière sur l'or de Guyane | Montant net de taxe minière sur l'or de Guyane": 502.08,
      "Taxe minière sur l'or de Guyane | Tarifs par kg extrait pour les PME": 498.06,
      "Taxe minière sur l'or de Guyane | Tarifs par kg extrait pour les autres entreprises": 0,
      "Total redevance des mines": 201.10999999999999,
    },
    Object {
      "Base des redevances | Nature": "Kilogramme d'or contenu",
      "Base des redevances | Quantités": "2.151",
      "Commune du lieu principal d'exploitation": "Sinnamary",
      "Désignation et adresse des concessionnaires, titulaires de permis d’exploitation ou exploitants": "titulaire2 - ladresse2 - legalSiren2 SIREN",
      "Frais de gestion de la fiscalité directe locale": 120.03759999999998,
      "Nature des substances extraites": "Minerais aurifères",
      "Numéro d'ordre de la matrice": 3,
      "Numéro de l'article du rôle": "slug-titre-2",
      "Redevance communale | Montant net redevance des mines": 357.72,
      "Redevance communale | Tarifs": 166.3,
      "Redevance départementale | Montant net": 71.41,
      "Redevance départementale | Tarifs": 33.2,
      "Service de la Direction générale des Finances publiques en charge du recouvrement": "Direction régionale des finances publiques (DRFIP) - Guyane",
      "Taxe minière sur l'or de Guyane | Montant des investissements déduits": 0,
      "Taxe minière sur l'or de Guyane | Montant net de taxe minière sur l'or de Guyane": 1071.34,
      "Taxe minière sur l'or de Guyane | Tarifs par kg extrait pour les PME": 498.06,
      "Taxe minière sur l'or de Guyane | Tarifs par kg extrait pour les autres entreprises": 0,
      "Total redevance des mines": 429.13,
    },
    Object {
      "Base des redevances | Nature": "Kilogramme d'or contenu",
      "Base des redevances | Quantités": "6.118",
      "Commune du lieu principal d'exploitation": "Saint-Laurent-du-Maroni",
      "Désignation et adresse des concessionnaires, titulaires de permis d’exploitation ou exploitants": "titulaire3 - ladresse3 - legalSiren3 SIREN",
      "Frais de gestion de la fiscalité directe locale": 1125.5896,
      "Nature des substances extraites": "Minerais aurifères",
      "Numéro d'ordre de la matrice": 4,
      "Numéro de l'article du rôle": "slug-titre-3",
      "Redevance communale | Montant net redevance des mines": 933.47,
      "Redevance communale | Tarifs": 166.3,
      "Redevance départementale | Montant net": 186.13,
      "Redevance départementale | Tarifs": 33.2,
      "Service de la Direction générale des Finances publiques en charge du recouvrement": "Direction régionale des finances publiques (DRFIP) - Guyane",
      "Taxe minière sur l'or de Guyane | Montant des investissements déduits": 5000,
      "Taxe minière sur l'or de Guyane | Montant net de taxe minière sur l'or de Guyane": 12950.27,
      "Taxe minière sur l'or de Guyane | Tarifs par kg extrait pour les PME": 498.06,
      "Taxe minière sur l'or de Guyane | Tarifs par kg extrait pour les autres entreprises": 0,
      "Total redevance des mines": 1119.6,
    },
  ],
  "matrice1122": Array [
    Object {
      "Communes sur le territoire desquels fonctionnent les exploitations": "Roura (1)",
      "Départements sur le territoire desquels fonctionnent les exploitations": "Guyane",
      "Désignation des concessionnaires": "titulaire1 - ladresse1 - legalSiren1 SIREN",
      "Désignation des concessions": "slug-titre-1",
      "Numéro d'ordre de la matrice": 1,
      "Observations": "production en kilogramme d'or",
      "Tonnages extraits ou cours de l'année précédente | par commune": "12.243",
      "Tonnages extraits ou cours de l'année précédente | par département": "12.243",
    },
    Object {
      "Communes sur le territoire desquels fonctionnent les exploitations": "Saint-Élie (0.3191007)",
      "Départements sur le territoire desquels fonctionnent les exploitations": "Guyane",
      "Désignation des concessionnaires": "titulaire2 - ladresse2 - legalSiren2 SIREN",
      "Désignation des concessions": "slug-titre-2",
      "Numéro d'ordre de la matrice": 2,
      "Observations": "production en kilogramme d'or",
      "Tonnages extraits ou cours de l'année précédente | par commune": "1.008",
      "Tonnages extraits ou cours de l'année précédente | par département": "1.008",
    },
    Object {
      "Communes sur le territoire desquels fonctionnent les exploitations": "Sinnamary (0.6808993)",
      "Départements sur le territoire desquels fonctionnent les exploitations": "Guyane",
      "Désignation des concessionnaires": "titulaire2 - ladresse2 - legalSiren2 SIREN",
      "Désignation des concessions": "slug-titre-2",
      "Numéro d'ordre de la matrice": 3,
      "Observations": "production en kilogramme d'or",
      "Tonnages extraits ou cours de l'année précédente | par commune": "2.151",
      "Tonnages extraits ou cours de l'année précédente | par département": "2.151",
    },
    Object {
      "Communes sur le territoire desquels fonctionnent les exploitations": "Saint-Laurent-du-Maroni (1)",
      "Départements sur le territoire desquels fonctionnent les exploitations": "Guyane",
      "Désignation des concessionnaires": "titulaire3 - ladresse3 - legalSiren3 SIREN",
      "Désignation des concessions": "slug-titre-3",
      "Numéro d'ordre de la matrice": 4,
      "Observations": "production en kilogramme d'or",
      "Tonnages extraits ou cours de l'année précédente | par commune": "6.118",
      "Tonnages extraits ou cours de l'année précédente | par département": "6.118",
    },
  ],
  "matrice1403": Array [
    Object {
      "Circonscription de": "SIP de Cayenne",
      "Nombre d'articles des rôles": 1,
      "Redevance communale": 284.71,
      "Redevance départementale": 56.27,
      "Sommes revenant au Conservatoire de biodiversité": 0,
      "Sommes revenant à l'État | Dégrèvements et non-valeurs": 0,
      "Sommes revenant à l'État | Frais d'assiette et de recouvrement": 79.4616,
      "Sommes revenant à l'État | Total": 79.4616,
      "Sommes revenant à la Région de Guyane": 652.29,
      "Taxe minière sur l'or de Guyane": 652.29,
      "Total des colonnes": 1072.7316,
    },
    Object {
      "Circonscription de": "SIP de Kourou",
      "Nombre d'articles des rôles": 2,
      "Redevance communale": 525.36,
      "Redevance départementale": 104.88,
      "Sommes revenant au Conservatoire de biodiversité": 0,
      "Sommes revenant à l'État | Dégrèvements et non-valeurs": 0,
      "Sommes revenant à l'État | Frais d'assiette et de recouvrement": 176.29279999999997,
      "Sommes revenant à l'État | Total": 176.29279999999997,
      "Sommes revenant à la Région de Guyane": 1573.4199999999998,
      "Taxe minière sur l'or de Guyane": 1573.4199999999998,
      "Total des colonnes": 2379.9528,
    },
    Object {
      "Circonscription de": "SIP de Saint-Laurent du Maroni",
      "Nombre d'articles des rôles": 1,
      "Redevance communale": 933.47,
      "Redevance départementale": 186.13,
      "Sommes revenant au Conservatoire de biodiversité": 0,
      "Sommes revenant à l'État | Dégrèvements et non-valeurs": 0,
      "Sommes revenant à l'État | Frais d'assiette et de recouvrement": 1125.5896,
      "Sommes revenant à l'État | Total": 1125.5896,
      "Sommes revenant à la Région de Guyane": 12950.27,
      "Taxe minière sur l'or de Guyane": 12950.27,
      "Total des colonnes": 15195.4596,
    },
  ],
  "matrice1404": Object {
    "cayenne": Array [
      Object {
        "Articles des rôles": 1,
        "Circonscription de": "SIP de Cayenne",
        "Communes": "Roura",
        "Départements": "Guyane",
        "Désignation des exploitants": "titulaire1 - ladresse1 - legalSiren1 SIREN",
        "Elements de base | Revenus imposables à la TFPB": 0,
        "Elements de base | Tonnages extraits": "12.243",
        "Redevance communale | Produit net de la redevance": 284.71,
        "Redevance communale | Revenant aux communes | 1ère fraction": 99.64849999999998,
        "Redevance communale | Revenant aux communes | 2ème fraction": 28.471,
        "Redevance communale | Revenant aux communes | Total": 128.1195,
        "Redevance communale | Répartition | 1ère fraction": 99.64849999999998,
        "Redevance communale | Répartition | 2ème fraction": 28.471,
        "Redevance communale | Répartition | 3ème fraction": 156.5905,
        "Redevance départementale | Produit net de la redevance": 56.27,
        "Redevance départementale | Sommes revenant aux départements": 56.27,
        "Taxe minière sur l'or de Guyane | Produit net": 652.29,
        "Taxe minière sur l'or de Guyane | Répartition | Conservatoire": 0,
        "Taxe minière sur l'or de Guyane | Répartition | Région de Guyane": 652.29,
      },
    ],
    "kourou": Array [
      Object {
        "Articles des rôles": 2,
        "Circonscription de": "SIP de Kourou",
        "Communes": "Saint-Élie",
        "Départements": "Guyane",
        "Désignation des exploitants": "titulaire2 - ladresse2 - legalSiren2 SIREN",
        "Elements de base | Revenus imposables à la TFPB": 0,
        "Elements de base | Tonnages extraits": "1.008",
        "Redevance communale | Produit net de la redevance": 167.64,
        "Redevance communale | Revenant aux communes | 1ère fraction": 58.67399999999999,
        "Redevance communale | Revenant aux communes | 2ème fraction": 16.764,
        "Redevance communale | Revenant aux communes | Total": 75.43799999999999,
        "Redevance communale | Répartition | 1ère fraction": 58.67399999999999,
        "Redevance communale | Répartition | 2ème fraction": 16.764,
        "Redevance communale | Répartition | 3ème fraction": 92.202,
        "Redevance départementale | Produit net de la redevance": 33.47,
        "Redevance départementale | Sommes revenant aux départements": 33.47,
        "Taxe minière sur l'or de Guyane | Produit net": 502.08,
        "Taxe minière sur l'or de Guyane | Répartition | Conservatoire": 0,
        "Taxe minière sur l'or de Guyane | Répartition | Région de Guyane": 502.08,
      },
      Object {
        "Articles des rôles": 3,
        "Circonscription de": "SIP de Kourou",
        "Communes": "Sinnamary",
        "Départements": "Guyane",
        "Désignation des exploitants": "titulaire2 - ladresse2 - legalSiren2 SIREN",
        "Elements de base | Revenus imposables à la TFPB": 0,
        "Elements de base | Tonnages extraits": "2.151",
        "Redevance communale | Produit net de la redevance": 357.72,
        "Redevance communale | Revenant aux communes | 1ère fraction": 125.202,
        "Redevance communale | Revenant aux communes | 2ème fraction": 35.772000000000006,
        "Redevance communale | Revenant aux communes | Total": 160.974,
        "Redevance communale | Répartition | 1ère fraction": 125.202,
        "Redevance communale | Répartition | 2ème fraction": 35.772000000000006,
        "Redevance communale | Répartition | 3ème fraction": 196.74600000000004,
        "Redevance départementale | Produit net de la redevance": 71.41,
        "Redevance départementale | Sommes revenant aux départements": 71.41,
        "Taxe minière sur l'or de Guyane | Produit net": 1071.34,
        "Taxe minière sur l'or de Guyane | Répartition | Conservatoire": 0,
        "Taxe minière sur l'or de Guyane | Répartition | Région de Guyane": 1071.34,
      },
    ],
    "saintLaurentDuMaroni": Array [
      Object {
        "Articles des rôles": 4,
        "Circonscription de": "SIP de Saint-Laurent du Maroni",
        "Communes": "Saint-Laurent-du-Maroni",
        "Départements": "Guyane",
        "Désignation des exploitants": "titulaire3 - ladresse3 - legalSiren3 SIREN",
        "Elements de base | Revenus imposables à la TFPB": 0,
        "Elements de base | Tonnages extraits": "6.118",
        "Redevance communale | Produit net de la redevance": 933.47,
        "Redevance communale | Revenant aux communes | 1ère fraction": 326.7145,
        "Redevance communale | Revenant aux communes | 2ème fraction": 93.34700000000001,
        "Redevance communale | Revenant aux communes | Total": 420.0615,
        "Redevance communale | Répartition | 1ère fraction": 326.7145,
        "Redevance communale | Répartition | 2ème fraction": 93.34700000000001,
        "Redevance communale | Répartition | 3ème fraction": 513.4085,
        "Redevance départementale | Produit net de la redevance": 186.13,
        "Redevance départementale | Sommes revenant aux départements": 186.13,
        "Taxe minière sur l'or de Guyane | Produit net": 12950.27,
        "Taxe minière sur l'or de Guyane | Répartition | Conservatoire": 0,
        "Taxe minière sur l'or de Guyane | Répartition | Région de Guyane": 12950.27,
      },
    ],
  },
}
`;
